apacheconfig/wloader.py

```diff
@@ -21,30 +23,90 @@ def _restore_original(word):
 @six.add_metaclass(abc.ABCMeta)
 class AbstractASTNode(object):
     """Generic class containing data that represents a node in the config AST.
+
+    There are three subclasses: :class:`apacheconfig.ListNode`,
+    :class:`apacheconfig.BlockNode`, and :class:`apacheconfig.LeafASTNode`.
+
+    Every AST should have a :class:`apacheconfig.ListNode` at its root. A
+    :class:`apacheconfig.ListNode` or :class:`apacheconfig.BlockNode` can have
+    other :class:`apacheconfig.BlockNode`s and
+    :class:`apacheconfig.LeafASTNode`s as children.
```

Comment:
nit: Is it deliberate to have *AST* in abstract and leaf, but not in other class names?